susePublicCloudInfoClient/lib/susepubliccloudinfoclient/infoserverrequests.py

```diff
@@ -60,29 +142,106 @@ def __get_base_url():
 def __get_data(url):
     """Make the request and return the data or None in case of failure"""
     response = requests.get(url)
+    print url
+    assert response.text, "No data was returned by the server!"
     return response.text
 
 
+def __inflect(plural):
+    inflections = {'images': 'image', 'servers': 'server'}
+    return inflections[plural]
+
+
+def __parse_command_arg_filter(command_arg_filter = None):
+    """Break down the --filter argument into a list of filters"""
+    valid_filters = {
+        'id': '^(?P<attr>id)(?P<operator>[=])(?P<value>.+)$',
+        'replacementid': '^(?P<attr>replacementid)(?P<operator>[=])(?P<value>.+)$',
+        'ip': '^(?P<attr>ip)(?P<operator>[=])(?P<value>\d+\.\d+\.\d+.\d+)$',
+        'name': '^(?P<attr>name)(?P<operator>[~])(?P<value>.+)$',
+        'replacementname': '(?P<attr>replacementname)(?P<operator>[~])(?P<value>.+)$',
+        'publishedon': '(?P<attr>publishedon)(?P<operator>[<=>])(?P<value>\d+)$',
+        'deprecatedon': '(?P<attr>deprecatedon)(?P<operator>[<=>])(?P<value>\d+)$',
+        'deletedon': '(?P<attr>deletedon)(?P<operator>[<=>])(?P<value>\d+)$'
+    }
+    # start with empty result set
+    filters = []
+    # split the argument into a comma-separated list if supplied...
+    if command_arg_filter:
+        for phrase in command_arg_filter.split(','):
+            # compare each comma-separated 'phrase' against the valid filters
+            # defined by regular expressions
+            for attr, regex in valid_filters.iteritems():
+                match = re.match(regex, phrase)
+                if match:
+                    filters.append(match.groupdict())
+                    break
+            else:
+                # if we can't break out with a valid filter, warn the user
+                __warn("Invalid filter phrase '%s' will be ignored." % phrase)
+    # return any valid filters we found
+    return filters
+
+
+def __parse_server_response_data(server_response_data, info_type):
+    return json.loads(server_response_data)[info_type]
+
+
+def __reformat(items, info_type, result_format):
+    if result_format == 'json':        
+        return json.dumps({info_type: items})
+    # default to XML output (until we have a plain formatter)
+    else:
+    # elif result_format == 'xml':
+        root = ET.Element(info_type)
+        for item in items:
+            ET.SubElement(root, __inflect(info_type), item)
+        return ET.tostring(root, 'UTF-8', 'xml')
+
+
+def __select_server_format(result_format, apply_filters=False):
+    if apply_filters:
+        return 'json'
+    elif result_format == 'plain':
+        return 'xml'
+    else:
+        return result_format
+ 
+
+def __warn(str, out=sys.stdout):
+    out.write("Warning: %s" % str)
```

Comment:
We need the warning, as previously agreed/discussed. I am just wondering if we should somehow make certain there is a separation to the results, similar argument than with the "print" statement above.